doc/nrf/glossary.rst

```diff
@@ -0,0 +1,404 @@
+.. _glossary:
+
+Glossary
+########
+
+.. contents::
+   :local:
+   :depth: 2
+
+Anycast addressing
+   An addressing type that routes datagrams to a single member of a group of potential receivers that are all identified by the same destination address.
+   This is a one-to-nearest association.
+
+Access Port Protection (APPROTECT)
+   A register used to prevent read and write access to all CPU registers and memory-mapped addresses.
+
+API Result
+   Indicates success or failure of an API procedure.
+   In case of failure, a comprehensive error code indicating the cause or reason for the failure is provided.
+
+Application
+   An implementation example that includes a variety of libraries to implement a specific use case.
+
+Application Programming Interface (API)
+   A language and message format used by an application program to communicate with an operating system, application, or other services.
+
+Board
+   In Zephyr and |NCS|, a data structure for describing hardware on the board and its configuration files.
+
+Carrier Wave (CW)
+   A single-frequency electromagnetic wave that can be modulated in amplitude, frequency, or phase to convey information.
+
+Cat-M1
+   LTE-M User Equipment (UE) category with a single RX antenna, specified in 3GPP Release 13.
```

Comment:
Should we link to the documents referenced on this page?

Reply:
I think @OisinFitz was for limiting the number of links in the page.